.k8s/K8s-Deployment.md

```diff
@@ -0,0 +1,45 @@
+# Kubernetes Deployment
+## Deployment
+### The Tool
+To work with a K8s cluster you need to install Kubectl, a tool for controlling your K8s deployment.
+[Kubectl installation guide](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+### Before Deploying
+Kubernetes manifests do not support adding config files directly in the manifest via a file path. Instead we need to use
+a resource called ConfigMaps.
+
+ConfigMaps are how we pass configurations to the containers. These can be created through literals or files. The only
+issue is that these can"t be created in a manifest based off a file like we used to do with Docker Compose. Now we have
+to created them manually using `kubectl create configmap <config-map-name> --from-file=<path-to-file>`
+
+### Deploying
+The deployment manifest file contains everything to deploy our application to the Kubernetes cluster. To deploy use:
+`kubectl apply -f search-deployment.yml`
+
+To remove a deployment and all related pods and containers use:
+`kubectl delete -f search-deployment.yml`
+
+Note: The ConfigMap will not be deleted automatically because it was added to the cluster manually.
+#### Optional:
+Install MiniKube, a tool for running Kubernetes locally in a VM or the host
+    machine. Deploys a lightweight single-node k8s cluster.
+Note: Run `eval $(minikube docker-env)` to enter Minikube"s docker context
+
+When starting up [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/), make sure your vm-driver is the one bundled with
```

Comment:
✏️ Probably put this link to install minikube at the first mention of minikube above rather than here.